From 3997ee9af83183b51ffadd6e6b0b4e0902779756 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?= <ablocorrea@hotmail.com>
Date: Thu, 14 Sep 2023 19:56:17 +0200
Subject: [PATCH] lib: remove gs-profiler.h from the public API

gs-profiler.h has the following include:

This breaks building OOT plugins, since first the file is not available,
and second, if it were, it might inject unwanted macros into the
external code.
---
 lib/gnome-software.h                 | 1 -
 lib/gs-plugin-job-refresh-metadata.c | 1 +
 lib/gs-plugin-loader.c               | 1 +
 lib/gs-profiler.h                    | 2 +-
 lib/gs-rewrite-resources.c           | 1 +
 lib/meson.build                      | 1 -
 plugins/flatpak/gs-flatpak.c         | 1 +
 plugins/flatpak/gs-plugin-flatpak.c  | 1 +
 8 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/lib/gnome-software.h b/lib/gnome-software.h
index 11f8e971f..c7e690802 100644
--- a/lib/gnome-software.h
+++ b/lib/gnome-software.h
@@ -38,7 +38,6 @@
 #include <gs-plugin-job-refresh-metadata.h>
 #include <gs-plugin-job-update-apps.h>
 #include <gs-plugin-vfuncs.h>
-#include <gs-profiler.h>
 #include <gs-remote-icon.h>
 #include <gs-rewrite-resources.h>
 #include <gs-utils.h>
diff --git a/lib/gs-plugin-job-refresh-metadata.c b/lib/gs-plugin-job-refresh-metadata.c
index 0802023cb..a53700c8d 100644
--- a/lib/gs-plugin-job-refresh-metadata.c
+++ b/lib/gs-plugin-job-refresh-metadata.c
@@ -50,6 +50,7 @@
 #include "gs-plugin-job-private.h"
 #include "gs-plugin-job-refresh-metadata.h"
 #include "gs-plugin-types.h"
+#include "gs-profiler.h"
 #include "gs-odrs-provider.h"
 #include "gs-utils.h"
 
diff --git a/lib/gs-plugin-loader.c b/lib/gs-plugin-loader.c
index c490bd7db..93b33f3e2 100644
--- a/lib/gs-plugin-loader.c
+++ b/lib/gs-plugin-loader.c
@@ -32,6 +32,7 @@
 #include "gs-plugin-event.h"
 #include "gs-plugin-job-private.h"
 #include "gs-plugin-private.h"
+#include "gs-profiler.h"
 #include "gs-utils.h"
 
 #define GS_PLUGIN_LOADER_UPDATES_CHANGED_DELAY	3	/* s */
diff --git a/lib/gs-profiler.h b/lib/gs-profiler.h
index fe4bcdfc3..a5be26676 100644
--- a/lib/gs-profiler.h
+++ b/lib/gs-profiler.h
@@ -11,7 +11,7 @@
 
 #pragma once
 
-#include <config.h>
+#include "config.h"
 #include <gnome-software.h>
 
 /**
diff --git a/lib/gs-rewrite-resources.c b/lib/gs-rewrite-resources.c
index 44f42657c..1155b867f 100644
--- a/lib/gs-rewrite-resources.c
+++ b/lib/gs-rewrite-resources.c
@@ -12,6 +12,7 @@
 #include <glib/gi18n.h>
 #include <gnome-software.h>
 
+#include "gs-profiler.h"
 #include "gs-rewrite-resources.h"
 
 /*
diff --git a/lib/meson.build b/lib/meson.build
index 9e85cf64e..35eb32a96 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -37,7 +37,6 @@ libgnomesoftware_public_headers = [
   'gs-plugin-loader-sync.h',
   'gs-plugin-types.h',
   'gs-plugin-vfuncs.h',
-  'gs-profiler.h',
   'gs-remote-icon.h',
   'gs-rewrite-resources.h',
   'gs-test.h',
diff --git a/plugins/flatpak/gs-flatpak.c b/plugins/flatpak/gs-flatpak.c
index e82a9faf9..e59579587 100644
--- a/plugins/flatpak/gs-flatpak.c
+++ b/plugins/flatpak/gs-flatpak.c
@@ -38,6 +38,7 @@
 #include "gs-flatpak.h"
 #include "gs-flatpak-transaction.h"
 #include "gs-flatpak-utils.h"
+#include "gs-profiler.h"
 
 struct _GsFlatpak {
 	GObject			 parent_instance;
diff --git a/plugins/flatpak/gs-plugin-flatpak.c b/plugins/flatpak/gs-plugin-flatpak.c
index 1582a4672..e56fa762b 100644
--- a/plugins/flatpak/gs-plugin-flatpak.c
+++ b/plugins/flatpak/gs-plugin-flatpak.c
@@ -36,6 +36,7 @@
 #include "gs-flatpak-transaction.h"
 #include "gs-flatpak-utils.h"
 #include "gs-metered.h"
+#include "gs-profiler.h"
 #include "gs-worker-thread.h"
 
 #include "gs-plugin-flatpak.h"
-- 
2.42.0

